test/integration/policyCompilerIntegration_test.go

```diff
@@ -0,0 +1,188 @@
+package integration
+
+import (
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path"
+	"runtime"
+	"strconv"
+	"strings"
+	"testing"
+
+	opa2 "github.com/Foundato/kelon/internal/pkg/opa"
+
+	"github.com/Foundato/kelon/configs"
+	dataInt "github.com/Foundato/kelon/internal/pkg/data"
+	requestInt "github.com/Foundato/kelon/internal/pkg/request"
+	translateInt "github.com/Foundato/kelon/internal/pkg/translate"
+	watcherInt "github.com/Foundato/kelon/internal/pkg/watcher"
+	"github.com/Foundato/kelon/pkg/api"
+	"github.com/Foundato/kelon/pkg/constants/logging"
+	"github.com/Foundato/kelon/pkg/data"
+	"github.com/Foundato/kelon/pkg/opa"
+	"github.com/Foundato/kelon/pkg/request"
+	"github.com/Foundato/kelon/pkg/translate"
+	"github.com/Foundato/kelon/pkg/watcher"
+	"gopkg.in/yaml.v3"
+)
+
+func TestMain(m *testing.M) {
+	exitVal := m.Run()
+	os.Exit(exitVal)
+}
+
+type PolicyCompilerTestEnvironment struct {
+	configWatcher  watcher.ConfigWatcher
+	policyCompiler opa.PolicyCompiler
+	t              *testing.T
+}
+
+func TestPolicyCompiler(t *testing.T) {
+	// change root path for files
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		t.Error(errors.New("error changing root for policyCompilerIntegration_test"))
+		t.FailNow()
+	}
+	dir := path.Join(path.Dir(filename), "../..")
+	err := os.Chdir(dir)
+	if err != nil {
+		panic(err)
+	}
+
+	// init configloader
+	configLoader := configs.FileConfigLoader{
+		DatastoreConfigPath: "./examples/local/config/datastore.yml",
+		APIConfigPath:       "./examples/local/config/api.yml",
+	}
```

Comment:
We might consider refactoring this logic into a more flexible testing suite (API we can use to setup Integration tests with multiple configurations and different inputs/outputs).

The current setup seems to just evaluate if the queries are always evaluated the same way, but we also might want to force some malicious behavior and see if our core logic really throws an error.

But this might be a further implementation step we should discuss with @containerpope first.

Reply:
Would be perfectly fine from my side. Due to the wrapper nature of our system, this is basically the only real business logic we have. Should be as much hardened as possible.

Reply:
I think we should go the little extra mile to refactor this logic into a little more reusable way with also the track of error tests (tests where a failure of the system is the expected outcome). I already had a call with @aobermair and he also seems to be optimistic about the implementation of this idea.

With this (a bit more generic) approach in place, hardening our system should just be a matter of work in terms of how much tests we want to write with this lightweight testing suite.